### ISSUE TITLE
iOS Problem with new version 3.5.2

### ISSUE THREAD

**Issue by problemadder** [TowelSniffer/Anki-Chess-2.0#51]
cool improvements over all. but sadly something is of now. best seen in the following pictures:
iPhone:
front is ok<img src="https://github.com/user-attachments/assets/440a2b4d-8ce5-4a16-bdc1-c42798236d9e" width="200" />
back is cut of:
<img src="https://github.com/user-attachments/assets/e9515507-264d-4867-a63c-f09bd208b96c" width="200" />
![IMG_0855](https://github.com/user-attachments/assets/78de4b6b-bdec-4575-b92a-b1d3d5377c3d)
![IMG_0856](https://github.com/user-attachments/assets/60b23830-b70f-4f15-ae24-14863db20c06)
iPad horizontal is fine:
![IMG_0858](https://github.com/user-attachments/assets/63f993dc-4f7f-48d5-b0ce-1e7368349de2)
![IMG_0857](https://github.com/user-attachments/assets/78850e00-7af5-4e69-994d-50cac2d983f1)
perhaps there is an easy fix.

**Comment by problemadder** [comment]
sry i dont know how to resize pictures in comments

**Comment by TowelSniffer** [comment]
anki mobile seems to have a bigger screen size than what is actually visible which is odd. First step is to try reducing the height for the ios section. Should be fine with that, as the cards seem to load fine otherwise. 

```
/* Styling for iPad and iPhone devices */
.iphone .iframe, .ipad .iframe {
  height: 85vh;
}

.iphone .body, .ipad .body {
  height: 85vh;
}
```

**Comment by TowelSniffer** [comment]
I would love to test with an Ios device. there is a lot of space up the top so I would be interested to know if there is any extra stuff happening on IOS. like some kind of hidden dive of a margin. If you can possibly use this to examine the html elements for me it would be a big help in me understanding what is happening here: [eruda console](https://github.com/ankidroid/Anki-Android/wiki/Development-Guide#via-eruda-console-for-mobile-browsers). I would love a screenshot of the following but on IOS. 

![image](https://github.com/user-attachments/assets/5b36dce6-1aa9-44fc-9752-4a82a59f6453)

I had a similar issue with a persistent margin on ankidroid, and it was sue to it having an extra container.

**Comment by TowelSniffer** [comment]
also when adjusting the height, if you change border:0 to border:1 in the styling section it will help you see what is going on:  you want the height to be the number at which the border just vanishes on the bottom and right. so on desktop, It happens at 100%:

99%:
![image](https://github.com/user-attachments/assets/c8f54686-d0f8-4f41-8bc0-dec44471bbc4)

100%:
![image](https://github.com/user-attachments/assets/c7a33073-e445-450f-bee4-c32076557af3)

I hope that makes sense

**Comment by problemadder** [comment]
hi and happy new year. I see a "eruda overlay" on mac - but dont see it on ios. so i cant trigger anything. I am going  to tinker with the css later on. 
even removing this completely does make no difference at all:
_/* Styling for iPad and iPhone devices */
.iphone .iframe, .ipad .iframe {
  height: 85vh;
}

.iphone .body, .ipad .body {
  height: 85vh;
}_

**Comment by problemadder** [comment]
One thing that stands out to me is that the board seems bigger on the back side. with tinkering i can change the iframe but the board seems too large

**Comment by TowelSniffer** [comment]
yeah that is very strange. You can try adding !important like so: 

```

/* Styling for iPad and iPhone devices */
.iphone .iframe, .ipad .iframe {
  height: 85vh !important;
}

.iphone .body, .ipad .body {
  height: 85vh !important;
}
```

or try 85% instead of vh

**Comment by TowelSniffer** [comment]
so Styling hasn't changed since 3.5.0.1. can you enable the border on the frame and send a picture of how it renders at 50vh or 50%, whatever works

**Comment by problemadder** [comment]
https://github.com/user-attachments/assets/d31969ec-8e69-4e51-b067-eb22c7111a18

**Comment by problemadder** [comment]
iframe {
  height: 97%;
  width: 97%;
  border: 1;
}

https://github.com/user-attachments/assets/7d1d460c-a5e5-4ed3-88d8-2d3970fc17a2

the board is somehow cut off on the left side

**Comment by TowelSniffer** [comment]
Can you try both the following with border

So like

```
height: 50% !important;
```

```
height: 50vh !important;
```

**Comment by problemadder** [comment]
I tried both. but it made no difference

**Comment by TowelSniffer** [comment]
Ok this is my fault, I am an idiot. I found a [great resource](https://gist.github.com/redoPop/10381ad5d576b9d73745) that allowed to to figure this out I'm pretty sure. 

it needs to be iframe not .iframe and body not .body

so
```
/* Styling for iPad and iPhone devices */
.iphone .iframe, .ipad .iframe {
  height: 85vh !important;
}

.iphone .body, .ipad .body {
  height: 85vh;
}
```

becomes 

```
/* Styling for iPad and iPhone devices */
.iphone iframe, .ipad iframe {
  height: 85vh;
}
```

can you test this with border and let me know the optimal height for max size. start with the following and let me know if the body can be left at 100%/100vh:
```
/* Styling for iPad and iPhone devices */
.iphone iframe, .ipad iframe {
  height: 85vh;
}

.iphone body, .ipad body {
  height: 100%;
}
```

**Comment by problemadder** [comment]
are you sure? https://docs.ankiweb.net/templates/styling.html#platform-specific-css

I tried it nevertheless (with % and vh and !important), but seems to have no impact at all?
But perhaps I am doing something wrong.


https://github.com/user-attachments/assets/480b4482-bead-4d67-b6c7-3f8c0649eeec

**Comment by TowelSniffer** [comment]
This is great. now the iframe height is actually reducing, I know how to fix this. I will fix the resizing on the viewer to allow the board to shrink further for smaller screens. 

Can you do me a big favor and tweek the height so that the sits just right inside the visible section. ignore the backside for now (I need to fix the html file for this) and just focus on the front. 

```
/* Styling for iPad and iPhone devices */
.iphone iframe, .ipad iframe {
  height: 85vh;
}

.iphone body, .ipad body {
  height: 100%;
}
```
this seemed to result in the card being below center which suggests the iframe is too tall. can you try leaving the body height at 100% as that seemed to not matter and lower the iframe height until the border becomes just visible. it looks like 85vh needs to be lowered a bit. 

You have been a big help btw thanks. once I know the correct height it should help me avoid similar issues in future.

**Comment by problemadder** [comment]
You're welcome!

like this? (50vh):

https://github.com/user-attachments/assets/d52f2498-1bda-4987-8681-b091acc7ea5a

**Comment by TowelSniffer** [comment]
Not exactly what I wanted, but it doesn't matter tbh as it will likely depend in screen size anyway. The goal is to have the iframe to be as big as it can get without leaving the bottom of the screen under the answer buttons. So at 50% it fits within the screen but doesn't take up the entire height, which means wasted space. This is mainly a concern with the backside as you want to be able to see as much as possible with the board moves. 

I made an estimate based in the videos you sent and a ruler, and It looks to be the sweet spot at about 77vh. Last release should display better. Let me know how you go with it.  I should be able to avoid similar issues in future. It's easier If I know an appropriate height for the iframe.

**Comment by problemadder** [comment]
looks great! (dont get irritated from background color)
no excuses for chess goals 2025 left ;)


https://github.com/user-attachments/assets/9bbae927-79ff-40de-b809-4dfe2ca3e1ea

**Comment by TowelSniffer** [comment]
Im quite happy with that. everything seems to be centering well.

**Comment by problemadder** [comment]
Yes and it looks great on iPad too. Worth the time! 👌Sent from my iPhoneOn 1. Jan 2025, at 16:37, TowelSniffer ***@***.***> wrote:﻿
Im quite happy with that. everything seems to be centering well.
